packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,427 @@
+    function minApprovals() public view virtual returns (uint256) {
+        return minApprovals_;
+    }
+
+    /// @notice Returns the number of approvals,
+    /// @param _proposalId The ID of the proposal.
+    /// @return The number of approvals.
+    function approvals(uint256 _proposalId) public view returns (uint256) {
+        return proposals[_proposalId].tally.approvals;
+    }
+
+    /// @notice Updates the minimal approval parameter.
+    /// @param _minApprovals The new minimal approval value.
+    function updateMinApprovals(uint256 _minApprovals)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        _updateMinApprovals(_minApprovals);
+
+        emit MinApprovalUpdated({minApprovals: _minApprovals});
+    }
+
+    /// @notice Adds new members to the address list and updates the minimum approval parameter.
+    /// @param _members The addresses of the members to be added.
+    function addAddresses(address[] calldata _members)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        _addAddresses(_members);
+
+        emit AddressesAdded({members: _members});
+    }
+
+    /// @notice Removes existing members from the address list. Previously, it checks if the new addresslist length at least as long as the minimum approvals parameter requires. Note that `minApprovals` is must be at least 1 so the address list cannot become empty.
+    /// @param _members The addresses of the members to be removed.
+    function removeAddresses(address[] calldata _members)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        _removeAddresses(_members);
+
+        // Check if the new addresslist has become shorter than the current minimum number of approvals required.
+        uint256 newAddresslistLength = addresslistLength();
+        if (newAddresslistLength < minApprovals_) {
+            revert MinApprovalsOutOfBounds({limit: newAddresslistLength, actual: minApprovals_});
+        }
+
+        emit AddressesRemoved({members: _members});
+    }
+
+    /// @notice Creates a new majority voting proposal.
+    /// @param _metadata The metadata of the proposal.
+    /// @param _actions The actions that will be executed after the proposal passes.
+    /// @param _approveProposal If `true`, the sender will approve the proposal.
+    /// @param _tryExecution If `true`, execution is tried after the vote cast. The call does not revert if early execution is not possible.
+    /// @return proposalId The ID of the proposal.
+    function createProposal(
+        bytes calldata _metadata,
+        IDAO.Action[] calldata _actions,
+        bool _approveProposal,
+        bool _tryExecution
+    ) external returns (uint256 proposalId) {
+        uint64 snapshotBlock = getBlockNumber64() - 1;
+
+        if (!isListedAtBlock(_msgSender(), snapshotBlock)) {
+            revert ProposalCreationForbidden(_msgSender());
+        }
```

Comment:
We should also have the settings to choose between:
- Anyone can create a proposal
- Only listed signers can create a proposal

```
        if (minProposerVotingPower() != 0 && !isListedAtBlock(_msgSender(), snapshotBlock)) {
            revert ProposalCreationForbidden(_msgSender());
        }
```